### ISSUE TITLE
Publisher Plugin Not Populating Code Analysis Widget (HTTP 500)

### ISSUE THREAD

**Issue by amateus1** [hygieia/hygieia#3461]
Hello. Help please, I'm using the Jenkins Publisher Plug in and the BUILD , ARTIFACT AND DEPLOY widget are showing data but the CODE ANALYSIS is not. The connection to Hygieia is successful and authorized. I have ran out of options. I believe the 500 error was resolved in another issue BY Satish but the resolution was not documented. Thank you in advanced for getting me unblocked.

Below is the Jinkins Job Log. There seems to be two issues:

Hygieia Test Publisher: Processing read error: /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target/surefire-reports/TEST-com.example.devops.DevopsApplicationTests.xml - even the the file is there and tries to process it.

hygieiaCodeQualityPublishStep throws Hygieia: Failed Publishing Complete Metric Data. Response Code: 500. Response Value= "Internal error."

Publishing data for API [http://mep-********eastus2.cloudapp.azure.com/api](http://mep-%2A%2A%2A%2A%2A%2A%2A%2Aeastus2.cloudapp.azure.com/api)
Hygieia: Published Build Data For Artifacts Publishing. Response Code: 201. Response Value= 62666f377fa7b700164886c4,6265f65f7fa7b7001709e279
Hygieia Build Artifact Publisher - Looking for file pattern '*.war' in directory /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target
Hygieia Artifact Publisher: Processing file: /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target/devops-1.8.0-SNAPSHOT.war
Hygieia: Published Build Artifact Data. Filename=devops-1.8.0-SNAPSHOT.war, Name=devops, Version=1.8.0-SNAPSHOT, Group=com.example.devops. Response Code: 201. Response Value= "6266705c7fa7b700164886c8"
[Pipeline] sh

echo **** STARTING PUBLISH STEP******
**** STARTING PUBLISH STEP******
[Pipeline] hygieiaTestPublishStep
Hygieia: Published Build Data For Test Publishing. Response Code: 201. Response Value= 62666f377fa7b700164886c4,6265f65f7fa7b7001709e279
Hygieia Test Result Publisher - Looking for file pattern 'TEST-*.xml' in directory /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target/surefire-reports
Hygieia Test Publisher: Processing file: /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target/surefire-reports/TEST-com.example.devops.DevopsApplicationTests.xml
Hygieia Test Publisher: Processing read error: /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target/surefire-reports/TEST-com.example.devops.DevopsApplicationTests.xml
Hygieia Test Publisher: Processing file: /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target/surefire-reports/TEST-com.example.devops.web.WelcomeControllerTest.xml
Hygieia Test Publisher: Processing read error: /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target/surefire-reports/TEST-com.example.devops.web.WelcomeControllerTest.xml
Hygieia: Published Test Data. Nothing to publish
[Pipeline] sh
echo **** STARTING CODE QUALITY PUBLISH STEP******
**** STARTING CODE QUALITY PUBLISH STEP******
[Pipeline] hygieiaCodeQualityPublishStep
Analysing 0 junit file(s)
Analysing 0 pmd file(s)
Analysing 0 findbugs file(s)
Analysing 0 checkstyle file(s)
Analysing 1 jacoco file(s)
Produced 6 metrics, publishing to Hygieia
**Hygieia: Failed Publishing Complete Metric Data. Response Code: 500. Response Value= "Internal error."**
[Pipeline] sh
echo **** STARTING SONAR PUBLISH STEP******
**** STARTING SONAR PUBLISH STEP******
[Pipeline] hygieiaSonarPublishStep
**Hygieia: Published Build Data For Sonar Publishing. Response Code: 201. Response Value=** 62666f377fa7b700164886c4,6265f65f7fa7b7001709e279
Hygieia: Published Sonar Result. Nothing to publish
[Pipeline] sh
echo **** STARTING DEPLOY PUBLISH STEP******
**** STARTING DEPLOY PUBLISH STEP******
[Pipeline] hygieiaDeployPublishStep
Publishing data for API http://mep-*********.com/api
Hygieia: Published Build Data For Deployment Publishing. Response Code: 201. Response Value= 62666f377fa7b700164886c4,6265f65f7fa7b7001709e279
Hygieia Deployment Publisher - Looking for file pattern '.war' in directory /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target
Hygieia Deployment Publisher: Processing file: /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target/devops-1.8.0-SNAPSHOT.war
Hygieia: Published Deploy Data: Response Code: 201. Response Value= "626670627fa7b700164886d3,62623aad2ab79c0019c4731e"
[Pipeline] hygieiaDeployPublishStep
Publishing data for API http://me********.azure.com/api
Hygieia: Published Build Data For Deployment Publishing. Response Code: 201. Response Value= 62666f377fa7b700164886c4,6265f65f7fa7b7001709e279
Hygieia Deployment Publisher - Looking for file pattern '.war' in directory /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target
Hygieia Deployment Publisher: Processing file: /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target/devops-1.8.0-SNAPSHOT.war
Hygieia: Published Deploy Data: Response Code: 201. Response Value= "626670637fa7b700164886d7,62623aad2ab79c0019c4731e"
[Pipeline] hygieiaDeployPublishStep
Publishing data for API http://me*********.com/api
Hygieia: Published Build Data For Deployment Publishing. Response Code: 201. Response Value= 62666f377fa7b700164886c4,6265f65f7fa7b7001709e279
Hygieia Deployment Publisher - Looking for file pattern '*.war' in directory /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target
Hygieia Deployment Publisher: Processing file: /var/lib/jenkins/jobs/devops-1-4-25-22/workspace/target/devops-1.8.0-SNAPSHOT.war
Hygieia: Published Deploy Data: Response Code: 201. Response Value= "626670637fa7b700164886db,62623aad2ab79c0019c4731e"
[Pipeline] sh

![165072239-16214400-61f8-44b0-8f0c-eabcca214ca3](https://user-images.githubusercontent.com/20121170/165073434-d17d7988-9d27-47b4-9494-bf735f08ddf6.png)
![image](https://user-images.githubusercontent.com/20121170/165073685-63f6b010-fda8-4a35-a1ef-edd20532a116.png)

![image](https://user-images.githubusercontent.com/20121170/165073711-b46df43f-34a1-4210-a216-d95865061f59.png)

**Comment by amateus1** [comment]
Anyone has any suggestions on how to fix this problem?   Greatly appreciate it
Is Hygieia no longer being supported?
